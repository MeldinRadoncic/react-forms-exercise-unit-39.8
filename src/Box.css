
.Box {
    border: 3px solid black;
    display: flex;
    justify-content: right;
    
}

.Box-btn {

}

.Box-btn:hover {
    background-color: red;
    color: white;
}