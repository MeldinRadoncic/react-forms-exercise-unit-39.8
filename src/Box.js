import React from "react";
import './Box.css'


const Box = ({id,color,width,height,handleRemove}) => {
    const removeBox = () => handleRemove(id)

    return (
        <div className="Box"
        id="box"
        data-testid="box"
        style={{
         backgroundColor:`${color}`,
        width:`${width}px`,
        height:`${height}px`
        }}
        
        >
            <button data-testid="removeBtn" id="removeBtn" className="Box-btn" onClick={removeBox} >X</button>
        </div>
    )
}



export default Box;