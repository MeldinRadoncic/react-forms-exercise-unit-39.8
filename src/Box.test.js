import { render } from "@testing-library/react";
import React from "react";
import Box from './Box.js'


it('Renders without crashing', () => {
    render(<Box/>)
})

it('Matches snapshot', () => {
    const {asFragment} = render(<Box/>)
    expect(asFragment()).toMatchSnapshot();
})