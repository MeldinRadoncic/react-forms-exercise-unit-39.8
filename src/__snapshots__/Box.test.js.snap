// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    data-testid="box"
    id="box"
  >
    <button
      class="Box-btn"
      data-testid="removeBtn"
      id="removeBtn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
